src/components/FlowRunTimelineSelectionPanel.vue

```diff
@@ -51,16 +53,42 @@
 
   const api = useWorkspaceApi()
   const routes = useWorkspaceRoutes()
-  const taskRunIdArgs = computed<[string] | null>(() => taskRunId.value ? [taskRunId.value] : null)
-  const taskRunSubscription = useSubscriptionWithDependencies(api.taskRuns.getTaskRun, taskRunIdArgs)
-  const taskRun = computed(() => taskRunSubscription.response)
+  const subscriptionArgs = computed<[string] | null>(() => id.value ? [id.value] : null)
 
-  const crumbs = computed(() => [
-    {
-      text: taskRun.value?.name ?? '',
-      to: taskRunId.value ? routes.taskRun(taskRunId.value) : '',
-    },
-  ])
+  let dataSubscription = assignSubscription()
+
+  function assignSubscription(): ReturnType<typeof useSubscriptionWithDependencies> {
+    return useSubscriptionWithDependencies(
+      type.value === 'task'
+        ? api.taskRuns.getTaskRun
+        : api.flowRuns.getFlowRun,
```

Comment:
Another example of what I mentioned above - this isn't a pattern I'd like to adopt.

Reply:
I'm not sure which previous comment you're referring to here?

Are you apposed to the conditional call type?

Reply:
https://github.com/PrefectHQ/prefect-ui-library/pull/1232#discussion_r1142677140

This one, sorry, I wrote these comments side by side then added more; we should have a really good motivation to use a ternary subscription and I don't think this rises to the level